packages/core/index.js

```diff
@@ -3,12 +3,19 @@ const fs = require("fs");
 const path = require("path");
 const { getWeb3 } = require("@uma/common");
 
+function getDirectoryForVersion(version) {
+  const packageName = `@uma/core-${version.split(".").join("-")}`;
+  return path.dirname(require.resolve(`${packageName}/package.json`));
+}
+
 /**
  * @notice Gets the truffle artifact for an UMA contract.
  * @param {String} contractName Name of the UMA contract whose artifact object will be returned.
+ * @param {String} [version] version identifier x.y.z for the contract. Defaults to "latest".
  */
-function getArtifact(contractName) {
-  const filePath = path.join(__dirname, "build", "contracts", `${contractName}.json`);
+function getArtifact(contractName, version = "latest") {
+  const contractDirectory = version === "latest" ? __dirname : getDirectoryForVersion(version);
+  const filePath = path.join(contractDirectory, "build", "contracts", `${contractName}.json`);
```

Comment:
One nice artifact of using the old build directories (and this technique in general) is that we still have access to old contract bytecode meaning we could use this same technique to deploy old contracts (even ones that we don't have the `.sol` for in the `contracts` directory.

Reply:
Yep! This might be really nice for testing backcompat.